### DISCUSSION TITLE
Images order in onDrop doesn't match user selection order

### DISCUSSION THREAD

**Discussion by FabioGodoyy** [react-dropzone/react-dropzone#1356] [General]
Hello,

I'm currently facing an issue with image order when using React Dropzone in my project. When the user selects several images to upload at once, they arrive in the onDrop event in alphabetical order, instead of following the order in which they were selected by the user.

This is causing significant inconvenience to our users as they spend considerable time manually rearranging images in the desired order after uploading.

I would like the images to arrive in the onDrop event in the same order in which they were selected by the user, so it will no longer be necessary to manually reorganize them.

Thanks in advance for any assistance or guidance in resolving this issue.

Steps to reproduce the issue:

- Open the image upload area.
- Select multiple images to upload in different orders.
- Note that images arrive in the onDrop event in random order, and not in the user's selection order.
- Additional Information:

React Dropzone version: ^14.2.2
Browser used: Google Chrome

**Comment by rolandjitsu** [comment]
@FabioGodoyy unfortunately, we do not enforce sorting on our end. That's most likely a browser limitation (I would imagine that it would be difficult for the browser to be able to tell the order in which you selected files on your system before you dropped them - in the case of a drag'n'drop).
